jujugui/static/gui/src/app/views/topology/relation.js

```diff
@@ -1159,9 +1159,8 @@ YUI.add('juju-topology-relation', function(Y) {
       var endpoint = Y.one(this).getAttribute('data-endpoint');
       var serviceId = endpoint.split(':')[0].trim();
       topo.fire('changeState', {
```

Comment:
In `environment.js` pass the changeState method into the `setAttrs` call in `createTopology` and then call it directly.